src/components/datepicker/datePicker.js

```diff
@@ -307,13 +444,30 @@
     var $scope = this.$scope;
     var keyCodes = this.$mdConstant.KEY_CODE;
 
-    // Add event listener through angular so that we can triggerHandler in unit tests.
-    self.ngInputElement.on('keydown', function(event) {
-      if (event.altKey && event.keyCode == keyCodes.DOWN_ARROW) {
-        self.openCalendarPane(event);
-        $scope.$digest();
-      }
-    });
+    if (!self.isDateRangePicker) {
+      // Add event listener through angular so that we can triggerHandler in unit tests.
+      self.ngInputElement.on('keydown', function(event) {
+        if (event.altKey && event.keyCode == keyCodes.DOWN_ARROW) {
```

Comment:
These `keydown` handlers are all exactly the same. They shouldn't be duplicated.